app/components/UserBets.tsx

```diff
@@ -243,7 +243,7 @@ export const UserBets = ({ fixedProductMarketMaker }: UserBets) => {
                         width="full"
                         size="lg"
                         className="space-x-2"
-                        onClick={redeem}
+                        onClick={redeem(index)}
```

Comment:
we need to change this to 
`() => redeem(index)`

Otherwise it will trigger the function when rendering the button vs when clicked.